homework/vector-of-shared-ptrs/vectorFunctions.cpp

```diff
@@ -0,0 +1,40 @@
+#include "vectorFunctions.hpp"
+#include <algorithm>
+#include <iostream>
+
+myVector generate(int count) {
+    myVector result(count);
+    std::for_each(begin(result), end(result),
+                  [number = 0](auto& pointer) mutable {
+                      pointer = std::make_shared<int>(number++);
+                  });
```

Comment:
This is a little to complicated for such simple thing :) Better use normal loop :) or std::generate